example_scenes/basic.py

```diff
@@ -178,4 +178,38 @@ def construct(self):
         self.wait()
 
 
+class VariableExample(Scene):
+    def construct(self):
+        var = 0.5
+        on_screen_var = Variable(var, "var", num_decimal_places=3)
+
+        # You can also change the colours for the label and value
+        on_screen_var.label.set_color(RED)
+        on_screen_var.value.set_color(GREEN)
+
+        self.play(Write(on_screen_var))
+        # The above line will just display the variable with
+        # its initial value on the screen. If you also wish to
+        # update it, you can do so by accessing the `tracker` attribute
+        self.wait()
+        var_tracker = on_screen_var.tracker
+        var = 10.5
+        self.play(var_tracker.set_value, var)
+        self.wait()
+
+        int_var = 0
+        on_screen_int_var = Variable(int_var, "int\_var", var_type=Integer).next_to(
```

Comment:
Ideally, user shouldn't have to provide the backslash in order to escape underscore, when they want the label `int_var`. Currently `Tex` is used for label. But an Exception is thrown: `Exception: LaTeX error converting to dvi. See log output above or the log file: media/Tex/d7548dde00f70cd3.log` if the underscore is not escaped.

Reply:
I believe this is because LaTeX itself requires the underscore to be escaped since it's usually used to denote a subscript. If the end user wants to have an underscore in their label, then yeah, they'll have to escape it.

Also, the reason the exception is thrown is because the subscript only works in LaTeX mathmode, so you might need to change the default label datatype from `Tex` to `MathTex`, or better yet, let the end user choose the label object they want to use, and if they pass a normal string, default it to `Tex`, `MathTex` or `Text`.

Reply:
Yeah letting the user choose is the way to go here

Reply:
I feel having to pass the class for the value would be a bit cumbersome:
`on_screen_var = Variable(var, "{a}_{i}", var_type=Integer, label_type=MathTex)`
Instead what I have tried in the latest commit is auto-detection of math mode, i.e if the user passes in a string wrapped in `$` (like "${a}_{i}$") then underscores aren't escaped and the label will be subscripted.
On the other hand, if something like `int_var` is passed, then underscores are escaped and `int_var` is displayed.

Reply:
Huh. I would much prefer if I could pass the MathTex/Tex/Text object _itself_ as a label.

Something like : `Variable(var, MathTex("{a}_{i}"), var_type=Integer)`. This way, we could allow for non-text based labels as well, like shapes.

If the user passes just a raw string as a label, like:
`Variable(var, "{a}_{i}", var_type=Integer)`, then it could default to MathTex, since I think those are the most frequently used Labelling objects.